src/blake2.rs

```diff
@@ -298,13 +301,14 @@ macro_rules! blake2_impl {
     }
 }
 
+#[cfg(test)]
 #[cfg_attr(feature = "clippy", allow(cast_possible_truncation))]
 #[cold]
-pub fn selftest_seq(len: usize) -> Vec<u8> {
-    use std::num::Wrapping;
+pub fn selftest_seq(len: usize) -> ::arrayvec::ArrayVec<[u8; 65536]> {
```

Comment:
The maximum number of elements used by the selftest is 1024; it might be better to change `test_write()` to also use 1024 bytes, instead of always allocating 64k on the stack here.

Reply:
Sure, should we make `test_write` just loop a thousand times or something like that?

Reply:
I don't think it's needed, since `test_write` is there just to check that `write` does the same as `update`. As long as the input is bigger than the hash block size, it should be enough. In this particular testcase, `selftest_seq` is just a source of pseudorandom data, what it generates doesn't matter as long as it's not a stream of zeros.